### PR TITLE
Decoupling Node and Manager

### PR DESCRIPTION
**Pull request by meling**
This PR aims to decouple the Node type from Manager.
There are still a few places where the dependence is tight, but perhaps this is ok.

### CODE REVIEW

File: cmd/benchmark/main.go [hunk 1]
```diff
@@ -130,20 +128,6 @@ func main() {
 		defer pprof.StopCPUProfile()
 	}
 
-	if *traceFile != "" {
```
Comment (Raytar) [inline cmd/benchmark/main.go:133]: Actually, this "tracing" code works fine. I was referring to the use of the `net/trace` package within Gorums. The `runtime/trace` package is different.

Comment (meling) [reply cmd/benchmark/main.go:133]: Okay. I see. Do you mean that `grpc.EnableTracing` can be removed along with `trace.IsEnabled()` check line 195.

File: mgr_test.go [hunk 2]
```diff
@@ -74,6 +79,39 @@ func TestManagerAddNode(t *testing.T) {
 		if err != nil && err.Error() != test.err {
 			t.Errorf("mgr.AddNode(%s, %d) = %s, expected %s", test.addr, test.id, err.Error(), test.err)
 		}
+	}
+}
 
+// Reusing tests/unresponsive/unresponsive.proto definitions
```
Comment (Raytar) [inline mgr_test.go:85]: Maybe we should have a dedicated package for this kind of testing?

File: cmd/benchmark/main.go (hunk 1 above)
Comment (Raytar) [reply cmd/benchmark/main.go:133]: Yes. But the `runtime/trace` stuff can stay

File: mgr_test.go (hunk 2 above)
Comment (meling) [reply mgr_test.go]: Added new `tests/dummy/dummy.proto` to be used for such testing. Please let me know if you have suggestions for a better name.

**Review by Raytar — APPROVED** [review]

**Pull request merged**